### PR TITLE
feat: add all recommended and must supported schema formats

### PR DESCRIPTION
**Pull request by jonaslagoni**
**Description**
This PR adds all the MUST and RECOMMENDED supported schema formats. This is to enable auto-completion more than anything because the underlying type is `string`, it's only to help know which known values are there.

### CODE REVIEW

**Comment by GreenRover** [comment]
If tooling will support proto but it is not listed here it is not a problem?

**Comment by jonaslagoni** [comment]
These are just all known schema formats listed in https://www.asyncapi.com/docs/reference/specification/v2.6.0#messageObjectSchemaFormatTable

We cannot add values we haven't agreed upon, you can of course still use custom ones.

Adding new keys for other schema formats is part of the spec discussion.

**Review by derberg — COMMENTED** [review]
looks like manual stuff that will have to be updated with each release

not a problem, just mentioning because this requires updates in readme, and probably script that generates new version definitions?

File: definitions/3.0.0/messageObject.json [hunk 1]
```diff
@@ -8,7 +8,57 @@
...
+            "application/vnd.aai.asyncapi+yaml;version=2.3.0",
+            "application/vnd.aai.asyncapi;version=2.4.0",
+            "application/vnd.aai.asyncapi+json;version=2.4.0",
+            "application/vnd.aai.asyncapi+yaml;version=2.4.0",
+            "application/vnd.aai.asyncapi;version=2.5.0",
+            "application/vnd.aai.asyncapi+json;version=2.5.0",
+            "application/vnd.aai.asyncapi+yaml;version=2.5.0",
+            "application/vnd.aai.asyncapi;version=2.6.0",
+            "application/vnd.aai.asyncapi+json;version=2.6.0",
+            "application/vnd.aai.asyncapi+yaml;version=2.6.0",
```
[the remark this answers is not in the document]
Comment (jonaslagoni) [reply definitions/3.0.0/messageObject.json:38]: There are two reasons why I am suggesting to have them:
- Code generation, i.e. that these types are generated and easier to use.
- Auto-completion, i.e. it's easier to remember the actual values and for the user.

From a validation point of view, they don't matter.

[the remark this answers is not in the document]
Comment (derberg) [reply definitions/3.0.0/messageObject.json:38]: but why below is not enough event for code gen and autocompletion, sorry but I don't get 😅 
```
"application/vnd.aai.asyncapi;version=3.0.0",
"application/vnd.aai.asyncapi+json;version=3.0.0",
"application/vnd.aai.asyncapi+yaml;version=3.0.0",
```

[the remark this answers is not in the document]
Comment (jonaslagoni) [reply definitions/3.0.0/messageObject.json:38]: He he, probably not 😄

I can remove previous versions 👍

**Comment by jonaslagoni** [comment]
> not a problem, just mentioning because this requires updates in readme, and probably script that generates new version definitions?

I'll see if I can include it in the script, that would be easiest yea.

**Comment by jonaslagoni** [comment]
Alright, I made a few changes.

1. Adapted the create new version script:
  - Before, we could accidentally replace versions that were unintended, i.e. it would happen in the next version (3.1.0) because it would replace the `application/vnd.oai.openapi;version=3.0.0` to `application/vnd.oai.openapi;version=3.1.0`. 
  - If the new version is a major version change (i.e. first number is not equal to the latest), replace all AsyncAPI schema format entries with the new.
  - If the new version is minor or fix, add a new entry of AsyncAPI schema format.
2. Adapted the root .asyncapi property to use const instead of enum. We will never have more then one value.
3. Removed all v2 schema formats

Depending on whether this or https://github.com/asyncapi/spec-json-schemas/pull/370 is merged first, we need to adapt the function `addNewSchemaVersion`.

**Review by fmvilas — CHANGES_REQUESTED** [review]

**Comment by jonaslagoni** [comment]
Applied your suggestions @fmvilas and applied the recent changes from master.

I adapted the add new version script so the validation of `schema` property is also changed accordingly with the right schema format values.

**Comment by jonaslagoni** [comment]
Please have another look @fmvilas @derberg 🙇

**Review by derberg — APPROVED** [review]
LGTM 🚀 

tested on local 👇🏼 

Screenshot 2023-07-25 at 14 55 28
Screenshot 2023-07-25 at 14 56 14

**Comment by derberg** [comment]
you mean this 👇

**Comment by derberg** [comment]
@fmvilas is out on holidays next 4 weeks, and for not PR is blocked as he requested changes to it in the past. 

I will use admin rights to dismiss his request for changes as we won't wait a month for him

**Pull request merged**